src/biip/gs1/_messages.py

```diff
@@ -78,6 +84,29 @@ def parse(
 
         return cls(value=value, element_strings=element_strings)
 
+    @classmethod
+    def parse_hri_string(value: str) -> GS1Message:
+        """Parse the GS1 string given in HRI (human readable interpretation) format.
+
+        Args:
+            value: The GS1 string to parse.
+
+        Returns:
+            Data class created by ``parse``.
+
+        Raises:
+            ParseError: If parsing of the data fails.
+        """
+        pattern = r"\((\d+)\)(\w+)"
+        matches = re.findall(pattern, value)
+        ai_map = {entry.ai: entry for entry in _GS1_APPLICATION_IDENTIFIERS}
```

Comment:
Instead of recreating this dict of >400 entries on each call to the method, feel free to change `_GS1_APPLICATION_IDENTIFIERS` to be a dict instead of a list.

Biip requires Python 3.7, so stable ordering of dicts is guaranteed. Existing use of `_GS1_APPLICATION_IDENTIFIERS` can be replaced with `_GS1_APPLICATION_IDENTIFIERS.values()`.

Reply:
Done.